plugins/provisioners/docker/client.rb

```diff
@@ -127,12 +127,11 @@ def container_args_changed?(config)
       def create_container(config)
         args = container_run_args(config)
 
-        @machine.communicate.sudo %[
-          rm -f #{config[:cidfile]}
-          docker run #{args}
-        ]
+        @machine.communicate.sudo %[rm -f #{config[:cidfile]}
+                                  ]
+        @machine.communicate.sudo %[docker run #{args}
+                                  ]
```

Comment:
It seems like we could still make this a single "command", so that vagrant isn't connecting twice to run the commands separately.

```ruby
 @machine.communicate.sudo %[rm -f #{config[:cidfile]}; docker run #{args}]
```

Maybe?